Feat/add support for filter data item value getter

anaivanovska opened the pull request.

Commit:
feat(grid): Add support for data item value getter in filter

Commit:
feat(grid): Use the data item value getter on address field

The addresses will be filter only by street, even though
the number is included in the column representation.

Commit:
feat(grid): Properly display filter on first/last columns

velinovskav commented on projects/data-grid/src/lib/models/data-grid-filter.ts:
```diff
@@ -25,12 +25,15 @@ export enum FilterType {
   value?: T;
   component?: Type<GridFilter>;
   props?: P;
+  dataItemValueGetter?: DataItemValueGetter;
```
Instead of defining a new property can't we use the GridDataFilterWithValueResolver?

Commit:
feat(grid): Use the existing GridDataFilterWithValueResolver

Commit:
feat(grid): Rename input argument

Commit:
feat(grid): Introduce directive responsible for filter reposition

- When view is initialized, adjust the position of the filter
if needed. If the origin element has width smaller than the filter element, then the offset on X should be smaller.
- Introduce needed Overlay module.
- Import the overlay style.

Commit:
feat(grid): Rename sortOrder variable

Commit:
feat(grid): Adjust the offset on y axis
